src/commands/external.rs

```diff
@@ -13,11 +15,30 @@ pub async fn execute_external_subcommand(args: Vec<String>) -> Result<()> {
     let plugin_name = args.first().ok_or_else(|| anyhow!("Expected subcommand"))?;
     let plugins_dir = get_spin_plugins_directory()?;
     check_plugin_spin_compatibility(plugin_name, env!("VERGEN_BUILD_SEMVER"), &plugins_dir)?;
-    let path = plugins_dir.join(plugin_name).join(plugin_name);
-    let mut command = Command::new(path);
+    let path = plugins_dir.join(plugin_name);
+    let binary = path.join(plugin_name);
+    let mut command = Command::new(binary);
     if args.len() > 1 {
         command.args(&args[1..]);
     }
+    let mut env_vars: HashMap<String, String> = HashMap::new();
```

Comment:
can creating the env_var map all be added to its own function? And maybe have an associated unit test if possible that checks that it is properly configured?